@xen-orchestra/lite/src/components/pool/dashboard/cpuUsage/VmsCpuUsage.vue

```diff
@@ -7,26 +7,30 @@
   </UsageBar>
 </template>
 <script lang="ts" setup>
-import { computed, onMounted, onUnmounted, ref, watchEffect } from "vue";
+import { computed, onMounted, onUnmounted } from "vue";
 import UsageBar from "@/components/UsageBar.vue";
 import useFetchStats from "@/composables/fetch-stats.composable";
-import { getAvgCpuUsage } from "@/libs/utils";
+import { deepComputed, getAvgCpuUsage } from "@/libs/utils";
 import { GRANULARITY, type VmStats } from "@/libs/xapi-stats";
-import type { XenApiVm } from "@/libs/xen-api";
 import { useVmStore } from "@/stores/vm.store";
 
 const vmStore = useVmStore();
 
-const runningVms = ref<XenApiVm[]>([]);
+const runningVms = deepComputed(() =>
+  vmStore.allRecords
+    .filter((vm) => vm.power_state === "Running")
+    .map((vm) => ({ uuid: vm.uuid, nameLabel: vm.name_label }))
+);
+
 const vmsWithStats = computed(() =>
-  runningVms.value.map((vm) => {
+  runningVms.value?.map((vm) => {
```

Comment:
`deepComputed` and `computed` should behave the same.

So here `runningVms.value` should always returns an array.

(See my proposal for `deepComputed` below)